### DISCUSSION TITLE
borders overwrites elements in table

### DISCUSSION THREAD

**Discussion by StinaRavdna** [ArthurSonzogni/FTXUI#910] [Q&A]
When I add border around the hbox/vbox the border overwrites the content of my tables.
How do I fix this? 
(I am a cpp and ftxui newbe)

When I run the code below in transparent view (diagnostic = false), I get the transparent view viewed perfectly with a border around the content. But when i run it in diagnostics view (diagnostic = true), the last element in all tables disappears when I keep this line : 
auto full_view = border(diagnostic_view);
When I remove it all content is shown, but then I do not have the border. 

My code:

int main() {
  using namespace ftxui;

  bool diagnostic = true;

    // Define menu table
    auto menu_table = Table({
      {"1", " - ", "Alt1"},
      {"2", " - ", "Alt2"},
      {"3", " - ", "Alt3"},
      {"4", " - ", "Alt4"},
      {"5", " - ", "Alt5"},
      {"6", " - ", "Alt6"},
      {"7", " - ", "Alt7"},
      {"8", " - ", "Alt8"},
      {"9", " - ", "Alt9"},
      {"e", " - ", "Exit"},
    });

    // Define diagnostics table
    auto diagnostics_table1 = Table({
      {"a:", " - ", "nnn"},
      {"b", " - ", "nnn"},
      {"c:", " - ", "nnn"},
      {"d", " - ", "nnn"},
      {"e", " - ", "nnn"},
      {"f", " - ", "nnn"},
      {"g", " - ", "nnn"},
      {"h", " - ", "nnn"},
    });

    auto diagnostics_table2 = Table({
      {"i", " - ", "nnn"},
      {"j", " - ", "nnn"},
      {"k", " - ", "nnn"},
      {"l", " - ", "nnn"},
      {"m", " - ", "nnn"},
      {"n", " - ", "nnn"},
      {"o", " - ", "nnn"},
      {"p", " - ", "nnn"},
    });
 
    auto menu_title = text("Options menu");
    auto status_title = text("Status");
    auto diagnostics_title = text("Diagnostics");

    //If diagnostic mode is not selected, show only the transparent view:
    auto transparent_view = ftxui::hbox({
      ftxui::vbox({
        menu_title,
        separator(),
        menu_table.Render(),
      }),
      separator(),
      ftxui::vbox({
        status_title,
        separator(),
        text("here comes a status overview"),
      })
    });
    
    auto diagnostic_tab = ftxui::vbox({
      diagnostics_title,
      separator(),
      ftxui::hbox({
        diagnostics_table1.Render(),
        separator(),
        diagnostics_table2.Render(),
        }),
    });

    auto diagnostic_view = ftxui::vbox({
      transparent_view,
      separator(),
      diagnostic_tab,
    });

    auto full_view = border(diagnostic_view);

    if (diagnostic) {
      ftxui::Screen diagnostic_screen = ftxui::Screen::Create(
        ftxui::Dimension::Fit(full_view),
          Dimension::Fit(full_view)
      );

      ftxui::Render(diagnostic_screen, full_view);
      diagnostic_screen.Print();
      std::cout << std::endl;

    } else {
        transparent_view = border(transparent_view);

        ftxui::Screen transparent_screen = ftxui::Screen::Create(
          ftxui::Dimension::Fit(transparent_view),
            Dimension::Fit(transparent_view)
        );

        ftxui::Render(transparent_screen, transparent_view);
        transparent_screen.Print();
        std::cout << std::endl;
    }
    
  return 0;
}

**Comment by ArthurSonzogni** [comment]
Hello, I am not sure to understand what you are reporting.
I observe:

### diagnostic = false
![image](https://github.com/user-attachments/assets/5894deed-c673-442a-85cd-3a1543d82a0f)

### diagnostic = true
![image](https://github.com/user-attachments/assets/e5cfbd23-3cce-42aa-aa56-2bdf363ff6a8)

What's wrong?

**Reply by StinaRavdna** [reply]
Weird. When I add the border, line "e - Exit" in the first table disappears and line "h - nnn" and "p -nnn" also disappears. Looks like the border takes up the space of the text. i am using vscode, might that cause some problems? I also get error messages like (see below) when I try to run example code, just copy-pasted as is.

/usr/bin/ld: main.cpp:(.text+0x873): undefined reference to `ftxui::Button(ftxui::ConstStringRef, std::function<void ()>, ftxui::ButtonOption)'
/usr/bin/ld: main.cpp:(.text+0x913): undefined reference to `ftxui::Button(ftxui::ConstStringRef, std::function<void ()>, ftxui::ButtonOption)'
/usr/bin/ld: main.cpp:(.text+0x9b3): undefined reference to `ftxui::Button(ftxui::ConstStringRef, std::function<void ()>, ftxui::ButtonOption)'

**Reply by StinaRavdna** [reply]
vscode on ubuntu

**Reply by StinaRavdna** [reply]
And thank you so much for answering!

**Reply by StinaRavdna** [reply]
Oh no, turned out my terminal window was too small! I'll open a new thread on the vscode question.

**Reply by ArthurSonzogni** [reply]
> Oh no, turned out my terminal window was too small! I

Lol ;-)
